# 1. Import all the necessary libraries



In [2]:
import pandas as pd
import numpy as np
import re
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns
import plotly.express as px
import emoji
import string
import nltk
from PIL import Image
from collections import Counter
from wordcloud import WordCloud, ImageColorGenerator, STOPWORDS
from nltk.tokenize import word_tokenize
from nltk.corpus import stopwords
from nltk.stem.porter import PorterStemmer
from nltk.stem.snowball import SnowballStemmer
from nltk.stem import WordNetLemmatizer

In [3]:
from sklearn.feature_extraction.text import CountVectorizer
from sklearn.feature_extraction.text import TfidfVectorizer
from sklearn.model_selection import train_test_split
from sklearn.model_selection import cross_val_score
from sklearn.metrics import accuracy_score, f1_score, confusion_matrix, classification_report
from sklearn.model_selection import GridSearchCV
from sklearn.model_selection import RandomizedSearchCV
from sklearn.linear_model import LogisticRegression
from sklearn.ensemble import RandomForestClassifier
from sklearn.ensemble import AdaBoostClassifier
from sklearn.svm import SVC,LinearSVC
from sklearn.naive_bayes import MultinomialNB
from sklearn.tree import DecisionTreeClassifier
from sklearn.pipeline import Pipeline
import pickle

In [4]:
import warnings
warnings.filterwarnings('ignore')

# 2. Import dataset from Kaggle

In [6]:
data = pd.read_csv('/Users/tativalentine/Documents/Capstone/cyberbullying_tweets.csv')

In [7]:
#Let's review the first 20 columns of our data in the dataset 
data.head(20)

tweet_text cyberbullying_type
0   In other words #katandandre, your food was cra...  not_cyberbullying
1   Why is #aussietv so white? #MKR #theblock #ImA...  not_cyberbullying
2   @XochitlSuckkks a classy whore? Or more red ve...  not_cyberbullying
3   @Jason_Gio meh. :P  thanks for the heads up, b...  not_cyberbullying
4   @RudhoeEnglish This is an ISIS account pretend...  not_cyberbullying
5   @Raja5aab @Quickieleaks Yes, the test of god i...  not_cyberbullying
6   Itu sekolah ya bukan tempat bully! Ga jauh kay...  not_cyberbullying
7   Karma. I hope it bites Kat on the butt. She is...  not_cyberbullying
8        @stockputout everything but mostly my priest  not_cyberbullying
9   Rebecca Black Drops Out of School Due to Bully...  not_cyberbullying
10               @Jord_Is_Dead http://t.co/UsQInYW5Gn  not_cyberbullying
11   The Bully flushes on KD http://twitvid.com/A2TNP  not_cyberbullying
12                                        Ughhhh #MKR  not_cyberbullying
13  RT @Kurdsnews: Turkish state has killed 241 ch...  not_cyberbullying
14  Love that the best response to the hotcakes th...  not_cyberbullying
15  @yasmimcaci @Bferrarii PAREM DE FAZER BULLYING...  not_cyberbullying
16  @sarinhacoral @Victor_Maggi tadinhu de mim , s...  not_cyberbullying
17  @0xabad1dea @kelseytheodore2 twitter is basica...  not_cyberbullying
18  Best pick up line? Hi, you're cute... ?: I lov...  not_cyberbullying
19  Now I gotta walk to classss?! I officially hat...  not_cyberbullying

In [8]:
# Pull out some data information (number of columns, data type and etc.)
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 47692 entries, 0 to 47691
Data columns (total 2 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   tweet_text          47692 non-null  object
 1   cyberbullying_type  47692 non-null  object
dtypes: object(2)
memory usage: 745.3+ KB


In [9]:
# Check for missing values
data.isnull().sum()

tweet_text            0
cyberbullying_type    0
dtype: int64

#### The provided data has no Null values!



In [11]:
# Next, let's show the count of unique values in the "cyberbullying_type" column
data['cyberbullying_type'].value_counts()

religion               7998
age                    7992
gender                 7973
ethnicity              7961
not_cyberbullying      7945
other_cyberbullying    7823
Name: cyberbullying_type, dtype: int64

The "cyberbullying_type" column of the dataset contains 6 unique values. The most frequent type of cyberbullying in the dataset is "religion", which occurs in 7998 instances. The second most frequent type is "age", which occurs in 7992 instances, followed by "gender" with 7973 instances. The least frequent type of cyberbullying in the dataset is "other_cyberbullying", which occurs in 7823 instances.

 This information provides an insight into the distribution of cyberbullying types in the dataset and can be helpful to understand the problem and design an appropriate ML model.
